src/pages/account/settings/index.tsx

```diff
@@ -0,0 +1,172 @@
+  dispatch: Dispatch<any>;
+  currentUser: CurrentUser;
+}
+
+type SettingsStateKeys = 'base' | 'security' | 'binding' | 'notification';
+interface SettingsState {
+  mode: 'inline' | 'horizontal';
+  menuMap: {
+    [key: string]: React.ReactNode;
+  };
+  selectKey: SettingsStateKeys;
+}
+@connect(({ accountAndsettings }: { accountAndsettings: { currentUser: CurrentUser } }) => ({
+  currentUser: accountAndsettings.currentUser,
+}))
+class Settings extends Component<SettingsProps, SettingsState> {
+  main: HTMLDivElement | undefined = undefined;
+
+  constructor(props: SettingsProps) {
+    super(props);
+    const menuMap = {
+      base: (
+        <FormattedMessage id="accountandsettings.menuMap.basic" defaultMessage="Basic Settings" />
+      ),
+      security: (
+        <FormattedMessage
+          id="accountandsettings.menuMap.security"
+          defaultMessage="Security Settings"
+        />
+      ),
+      binding: (
+        <FormattedMessage
+          id="accountandsettings.menuMap.binding"
+          defaultMessage="Account Binding"
+        />
+      ),
+      notification: (
+        <FormattedMessage
+          id="accountandsettings.menuMap.notification"
+          defaultMessage="New Message Notification"
+        />
+      ),
+    };
+    this.state = {
+      mode: 'inline',
+      menuMap,
+      selectKey: 'base',
+    };
+  }
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'accountAndsettings/fetchCurrent',
+    });
+    window.addEventListener('resize', this.resize);
+    this.resize();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  getMenu = () => {
+    const { menuMap } = this.state;
+    return Object.keys(menuMap).map(item => <Item key={item}>{menuMap[item]}</Item>);
+  };
+
+  getRightTitle = () => {
+    const { selectKey, menuMap } = this.state;
+    return menuMap[selectKey];
+  };
+
+  selectKey = (key: SettingsStateKeys) => {
+    this.setState({
+      selectKey: key,
+    });
+  };
+
+  resize = () => {
+    if (!this.main) {
+      return;
+    }
+    requestAnimationFrame(() => {
+      if (!this.main) {
+        return;
+      }
+      let mode: 'inline' | 'horizontal' = 'inline';
+      const { offsetWidth } = this.main;
+      if (this.main.offsetWidth < 641 && offsetWidth > 400) {
+        mode = 'horizontal';
+      }
+      if (window.innerWidth < 768 && offsetWidth > 400) {
+        mode = 'horizontal';
+      }
+      this.setState({
+        mode,
+      });
+    });
+  };
+
+  renderChildren = () => {
+    const { selectKey } = this.state;
+    switch (selectKey) {
+      case 'base':
+        return <BaseView />;
+      case 'security':
+        return <SecurityView />;
+      case 'binding':
+        return <BindingView />;
+      case 'notification':
+        return <NotificationView />;
+      default:
+        break;
+    }
+
+    return null;
+  };
+
+  render() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class method 'render' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)